File: ArraySmallest.java
import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

/* This program would find and display the smallest number in a
list of numbers that the user inputs */

public class ArraySmallest {
    public static void main(String[] sigma) {
        Scanner sc = new Scanner(System.in); 
        List<Integer> numbers = new ArrayList<>();
        int smallest = Integer.MAX_VALUE;
        // Prompt the user to enter numbers and add them to the list
        
        System.out.println("Enter the numbers one at a time."
        + " Enter 'Done' or press Enter on an empty line when you are done entering numbers.");

        while (true) {
            System.out.print("> ");
            // int num = scanner.nextInt();
            String input = sc.nextLine();
            if (input.equalsIgnoreCase("Done") || input.isEmpty()) {
                sc.close();
                break;
            }
            int num = Integer.parseInt(input);
            numbers.add(num);
        }

        // Find the smallest number in the list
        if (!numbers.isEmpty()) {
            for (int num : numbers) {
                if (num < smallest) {
                    smallest = num;
                }   
            }
        } else {
            System.out.println("No numbers were entered.");
            return;
        }

        // Display the smallest number
        System.out.println("\nThe smallest number is: " + smallest);
    }
}

File: Python/ArraySmallest.py
def find_smallest_number():
    """
    Prompts the user to enter numbers, then finds and prints the smallest one.
    """
    numbers = []
    
    print("Enter numbers one at a time.")
    print("Enter 'Done' or press Enter on an empty line when you are finished.")

    while True:
        ui = input("> ")
        
        # Check for termination condition
        if ui.lower() == 'done' or ui == "": 
        # checks if the user input (ui) is 'done' (case insensitive) or an empty string.
            break
        
        # Convert input to an integer and add to the list
        try:
            num = int(ui)
            numbers.append(num)
        except ValueError:
            # Handle cases where the input is not a valid number
            print("Invalid input. Please enter a number, 'Done', or an empty line.")

    # Find and display the smallest number
    if numbers:
        smallest = min(numbers) 
        # min() function returns the smallest item in an iterable or the smallest of two or more arguments.
        
        print(f"\nThe smallest number is: {smallest}")
    else:
        print("\nNo numbers were entered.")

# Run the main function when the script is executed
if __name__ == "__main__":  # This condition checks if the script is being run directly by the Python interpreter.
    find_smallest_number() # If the script is being run directly, it calls the main function to start the program.